src/main/java/com/google/cloud/tools/managedcloudsdk/command/AsyncCommandWrapper.java

```diff
@@ -14,52 +14,59 @@
  * limitations under the License.
  */
 
-package com.google.cloud.tools.managedcloudsdk.gcloud;
+package com.google.cloud.tools.managedcloudsdk.command;
 
-import com.google.cloud.tools.managedcloudsdk.MessageListener;
 import com.google.cloud.tools.managedcloudsdk.executors.SdkExecutorServiceFactory;
 import com.google.cloud.tools.managedcloudsdk.executors.SingleThreadExecutorServiceFactory;
-import com.google.cloud.tools.managedcloudsdk.process.CommandRunner;
 import com.google.common.util.concurrent.ListenableFuture;
 import com.google.common.util.concurrent.ListeningExecutorService;
-import java.nio.file.Path;
-import java.util.List;
 import java.util.concurrent.Callable;
 
-/** Wrapper around {@link CommandRunner} command to run gcloud commands asynchronously. */
-public class AsyncGcloudRunnerWrapper {
+/** Wrapper around {@link CommandRunner}/{@link CommandCaller} to asynchronous execution. */
+public class AsyncCommandWrapper {
 
-  private final GcloudCommandFactory commandFactory;
   private final SdkExecutorServiceFactory executorServiceFactory;
 
-  AsyncGcloudRunnerWrapper(
-      GcloudCommandFactory commandFactory, SdkExecutorServiceFactory executorServiceFactory) {
-    this.commandFactory = commandFactory;
+  AsyncCommandWrapper(SdkExecutorServiceFactory executorServiceFactory) {
     this.executorServiceFactory = executorServiceFactory;
   }
 
-  /** Run a command and return a resultless control future. */
-  public ListenableFuture<Void> runCommand(
-      final List<String> parameters, final MessageListener messageListener) {
+  /** Wrap a command runner and return a resultless future. */
+  public ListenableFuture<Void> run(final CommandRunner commandRunner) {
```

Comment:
This can probably be refactored into one function that just gets called with the different Callables.